shared/views/RoomDirectoryView.js

```diff
@@ -51,13 +47,37 @@ class RoomDirectoryView extends TemplateView {
         },
       },
       [
-        ...availableHomeserverOptionElements,
+        t.map(
+          (vm) => vm.availableHomeserverList,
+          (_, t, vm) => {
+            const availableHomeserverOptionElements = vm.availableHomeserverList.map(
+              (homeserverName) => {
+                return t.option(
+                  { value: homeserverName, selected: vm.homeserverSelection === homeserverName },
+                  homeserverName
+                );
+              }
+            );
+
+            let availableHomeserversOptGroup = text('');
+            if (availableHomeserverOptionElements.length > 0) {
+              availableHomeserversOptGroup = t.optgroup(
+                { label: 'Defaults' },
+                availableHomeserverOptionElements
+              );
```

Comment:
Have to put this under a "Defaults" `<optgroup>` in order to be compatible with Hydrogen TemplateView's requiring a single root element.

I would ideally just return the array of `<option>` elements as before but we need the data binding to update the list when it changes. I don't want to use a `ListView` because it's cumbersome with it's observable list requirement.